sdk/python/kfp/dsl/for_loop.py

```diff
@@ -160,11 +179,16 @@ def from_pipeline_channel(
     ) -> 'LoopArgument':
         """Creates a LoopArgument object from a PipelineParameterChannel
         object."""
+        if channel.channel_type.startswith('typing.Dict'):
+            loop_argument_channel_type = channel.channel_type
```

Comment:
What happens if the user annotates `-> dict` with no inner types?

This commented is related to the other comment about unit testing the different cases we might encounter.